src/main/java/com/atlassian/migration/datacenter/core/fs/S3FilesystemMigrationService.java

```
@@ -69,6 +80,49 @@ public FileSystemMigrationReport getReport() {
         return report;
     }
 
+    @Override
+    public Boolean scheduleMigration() {
+        Migration currentMigration = this.migrationService.getCurrentMigration();
+        //TODO: Review why this check is needed?
+        if (currentMigration.getStage() == NOT_STARTED) {
+            return false;
+        }
+
+        final JobRunnerKey runnerKey = JobRunnerKey.of(S3UploadJobRunner.KEY);
+        JobId jobId = JobId.of(S3UploadJobRunner.KEY + currentMigration.getID());
+        logger.info("Starting filesystem migration");
+
+        //TODO: Can the job runner be injected? It has no state
+        schedulerService.registerJobRunner(runnerKey, new S3UploadJobRunner(this));
+        logger.info("Registered new job runner for S3");
+
+//      TODO: don't schedule when the job is running, there are few edge cases we need to cover
+//        JobDetails jobDetails = schedulerService.getJobDetails(jobId);
+//        if (jobDetails != null) {
+//            final Date nextRunTime = jobDetails.getNextRunTime();
+//            if (nextRunTime != null && nextRunTime.before(new Date())) {
+//                return false;
+//            }
+//        }
+
+        JobConfig jobConfig = JobConfig.forJobRunnerKey(runnerKey)
+                .withSchedule(null) // run now
+                .withRunMode(RunMode.RUN_ONCE_PER_CLUSTER);
+        try {
+            logger.info("Scheduling new job for S3 upload runner");
+
+            this.migrationService.transition(MigrationStage.FS_MIGRATION_COPY, MigrationStage.WAIT_FS_MIGRATION_COPY);
+
+            schedulerService.scheduleJob(jobId, jobConfig);
+        } catch (SchedulerServiceException | InvalidMigrationStageError e) {
```

I think we can throw the `InvalidMigrationStageError` exception from here. We can also create `FilesystemMigrationException` to wrap any other errors errors.